java/src/shared/models/game/GameMap.java

```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
         this.cities = cities;
     }
 
+    /**
+     * Constructor that creates GameMap based on if the user wants random hex types, random port types,
+     * and/or random numbers on hexes
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomPorts   whether the Port Types are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public GameMap(boolean randomTiles, boolean randomPorts, boolean randomNumbers) {
+        setNewGameHexes(randomTiles, randomNumbers);
```

Comment:
Are we ensuring that the desert hex always gets no number assigned, and that every other hex gets a valid number?